src/main/scala/io/iohk/ethereum/mallet/service/RpcClientMallet.scala

```diff
@@ -81,15 +81,17 @@ class RpcClientMallet(node: Uri)(implicit system: ActorSystem, ec: ExecutionCont
     val entity = HttpEntity(ContentTypes.`application/json`, jsonRequest.noSpaces)
     val request = HttpRequest(method = HttpMethods.POST, uri = node, entity = entity)
 
-    val responseF: Future[Either[Err, Json]] = Http()
-      .singleRequest(request)
-      .flatMap(_.entity.toStrict(httpTimeout))
-      .map(e => parse(e.data.utf8String).left.map(e => RpcClientError(e.message)))
-      .recover { case ex =>
-        Left(RpcClientError("RPC request failed: " + Util.exceptionToString(ex)))
-      }
-
-    Try(Await.result(responseF, httpTimeout)) match {
+    val responseF: Task[Either[Err, Json]] =
+      Task
+        .fromFuture(Http().singleRequest(request))
+        .map(_.entity.toStrict(httpTimeout))
+        .flatMap(Task.fromFuture)
```

Comment:
why we use here Task.fromFuture ?

Reply:
The problem is that `Http().singleRequest(request)` returns a `Future`, and then `_.entity.toStrict(httpTimeout)` also returns a `Future`, so basically I'm wrapping Futures twice :/ Better ideas are more than welcome.